_extras/survey-templates.md

```diff
@@ -0,0 +1,75 @@
+
+- You are provided with a CSV file. The file has 35000 rows. The file has 45 columns. Each column is separated by the “,” symbol from the other. You would like to open the file in python, remove all entries where the value of column 21 is larger than 50. The values removed are to be replaced by 0.
+
+- You are provided with a CSV file. The file has 35000 rows. The file has 45 columns. Each column is separated by the “,” symbol from the other. When you load the file and plot the histogram of column 40, you are suspicious that the floating point values are not normally distributed. But, the producer of the CSV file assures you that all columns are normally distributed. To make sure, you sit down to code a function which tests any given column if it is normally distributed.
+
+- A probability density function (PDF) in statistics is something very distinct from a probability mass function (PMF). What is the difference?
+
+- You are given samples of two sets of observations with type float32. Both arrays have a similar shape: "(1024, 1)" You'd like to compare both ensembles if they are produced by the same PDF. How do you do this?
+
+### Questions about machine learning
+
+**The following questions range from simple to more advanced.**
+
+- You are given a dataset from experiments that you want to use for machine learning (13 columns with 25000 rows). One column is particularly useful and is encoded as real numbers in a range from -15 to 12. You would like to normalize this data so that it fits into the range of real numbers between 0 and 1. How would you do this?
+
+- You are helping to organize a conference of more than 1000 attendants. All participants have already paid and are expecting to pick up their conference t-shirt on the first day. Your team is in shock as it discovers that t-shirt sizes have not been recorded during online registration. However, all participants were asked to provide their age, gender, body height and weight. To help out, you sit down to write a python script that predicts the t-shirt size for each participant using a clustering algorithm.
+
+- You are presented a training data set of 10000 samples. You would like to train a classifier on this datasets. However, you observe that class 0 dominates the training set at 60%, the other classes equally share 20% of the rows each. How do you prepare your training procedure?
+
+-  You are given a classifier by a colleague. Your peer trained it on a very large dataset. You just need it for inference. You reassure yourself if the model you loaded from disk really reproduces the weights by running the 'predict' function on an unseen validation set by your colleague. The network doesn't make the same predictions as your colleague told you. What could be the cause?
```

Comment:
"The ~network~ model doesn't make the same predictions as your colleague told you." One could specify which data set
the colleague used to create predictions that now cannot be reproduced.

Reply:
I adapted the wording a bit. 9f93d49 
What do you think?

I'd not be over-precise on the dataset question you raised so that people are not too confused. Or maybe I mistake what you mean. Could you provide an example, please?

Reply:
Hmm, I understood:

* colleague trains model unsing `X_train`, `Y_train`
* tells me "result should be `Y_test`", doesn't tell me input is `X_test`
* I use `Y_valid = model.predict(X_valid) != Y_test`